notifications/src/main/java/io/castled/CastledNotifications.kt

```diff
@@ -0,0 +1,223 @@
+            if (checkSdkInitialized("logAppOpenedEvent"))
+                inApp.logAppOpenedEvent(activity)
+        }
+
+        @JvmStatic
+        fun logAppOpenedEvent(context: Context) {
+            if (checkSdkInitialized("logAppOpenedEvent"))
+                inApp.logAppOpenedEvent(context)
+        }
+
+
+        /**
+         * CustomEvent tracked by SDk with overloaded methods
+         */
+        @JvmStatic
+        fun logCustomEvent(
+            appCompatActivity: AppCompatActivity,
+            eventName: String,
+            eventParams: Map<String, Any>
+        ) {
+            if (checkSdkInitialized("logCustomEvent"))
+                inApp.logCustomEvent(appCompatActivity, eventName, eventParams)
+        }
+
+        @JvmStatic
+        fun logCustomEvent(
+            activity: Activity,
+            eventName: String,
+            eventParams: Map<String, Any>
+        ) {
+            if (checkSdkInitialized("logCustomEvent"))
+                inApp.logCustomEvent(activity, eventName, eventParams)
+        }
+
+        @JvmStatic
+        fun logCustomEvent(
+            context: Context,
+            eventName: String,
+            eventParams: Map<String, Any>
+        ) {
+            if (checkSdkInitialized("logCustomEvent"))
+                inApp.logCustomEvent(context, eventName, eventParams)
+        }
+
+
+        /**
+         * LogEvent only by event name with overloaded methods
+         */
+        @JvmStatic
+        fun logEvent(appCompatActivity: AppCompatActivity, eventName: String) {
+            if (checkSdkInitialized("logEvent"))
+                inApp.logEvent(appCompatActivity, eventName)
+        }
+
+        @JvmStatic
+        fun logEvent(activity: Activity, eventName: String) {
+            if (checkSdkInitialized("logEvent"))
+                inApp.logEvent(activity, eventName)
+        }
+
+        @JvmStatic
+        fun logEvent(context: Context, eventName: String) {
+            if (checkSdkInitialized("logEvent"))
+                inApp.logEvent(context, eventName)
+        }
+
+
+        /**
+         * LogEvent with Event name and with Event param with overloaded methods
+         */
+        @JvmStatic
+        fun logEvent(
+            appCompatActivity: AppCompatActivity,
+            eventName: String,
+            eventParams: Map<String, Any>
+        ) {
+            if (checkSdkInitialized("logEvent"))
+                inApp.logEvent(appCompatActivity, eventName, eventParams)
+        }
+
+        @JvmStatic
+        fun logEvent(activity: Activity, eventName: String, eventParams: Map<String, Any>) {
+            if (checkSdkInitialized("logEvent"))
+                inApp.logEvent(activity, eventName, eventParams)
+        }
+
+        @JvmStatic
+        fun logEvent(context: Context, eventName: String, eventParams: Map<String, Any>) {
+            if (checkSdkInitialized("logEvent"))
+                inApp.logEvent(context, eventName, eventParams)
+        }
```

Comment:
Both me and my dev's are aware that context is a super class of activity and appcompatactivity. These overloaded methods were used to use some activity lifecycle observer methods for automatic tracking code that has now been removed. Not all methods on activity, or appcompatactivity are available on context as you must be aware. We chose to use an overloaded method instead of an instanceof check as it's more type safe. We retained the methods until we e2e check automatic activity tracking. We can remove these methods, we might recreate something similar in future if required